### PR TITLE
Clicking cancel on the carp rift will no longer summon you as a carp

### LINKED ISSUE #64572: Clicking cancel on space dragon "become a carp" prompt still makes you a carp
## Reproduction:

1. Click space carp rift
2. Click cancel
3. Become fish

@jlsnow301

### PR DESCRIPTION
**Pull request by ghost**
## About The Pull Request

Clicking cancel on the carp rift will no longer summon players as carps

## Why It's Good For The Game

Bug fix

## Changelog

:cl:
fix: Clicking cancel on the carp rift will no longer summon you as a carp
/:cl:

### CODE REVIEW

**Comment by SuperNovaa41** [comment]
Fixes https://github.com/tgstation/tgstation/issues/64572

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 1]
```diff
@@ -696,7 +698,7 @@
 			return FALSE
 		is_listed = TRUE
 	var/carp_ask = tgui_alert(usr,"Become a carp?", "Help bring forth the horde?", list("Yes", "No"))
-	if(carp_ask == "No" || !src || QDELETED(src) || QDELETED(user))
+	if(!(carp_ask == "Yes") || !src || QDELETED(src) || QDELETED(user))
```
Comment (Fikou) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:701]:
```suggestion
	if(carp_ask != "Yes" || !src || QDELETED(src) || QDELETED(user))
```

File: code/modules/mob/living/simple_animal/hostile/space_dragon.dm [hunk 2]
```diff
@@ -696,7 +698,7 @@
 			return FALSE
 		is_listed = TRUE
 	var/carp_ask = tgui_alert(usr,"Become a carp?", "Help bring forth the horde?", list("Yes", "No"))
```
Comment (jlsnow301) [inline code/modules/mob/living/simple_animal/hostile/space_dragon.dm:700]:
```suggestion
	var/carp_ask = tgui_alert(user, "Become a carp?", "Carp Rift", list("Yes", "No"))
```
L + message / title ratio + spacing + user

Comment (Sealed101) [reply code/modules/mob/living/simple_animal/hostile/space_dragon.dm:700]: dont touch the message its funny even if it isnt as true anymore

Comment (jlsnow301) [reply code/modules/mob/living/simple_animal/hostile/space_dragon.dm:700]: Id rather they be swapped than the title longer than the message

Comment (jlsnow301) [reply code/modules/mob/living/simple_animal/hostile/space_dragon.dm:700]: The message is the second argument, for context

**Pull request merged**